x/wasm/client/cli/genesis_msg.go

```diff
@@ -377,44 +397,52 @@ func readGenesis(cmd *cobra.Command) (*genesisData, error) {
 		clientCtx.JSONMarshaler.MustUnmarshalJSON(appState[types.ModuleName], &wasmGenesisState)
 	}
 
-	return &genesisData{
-		genesisFile:     genFile,
-		genDoc:          genDoc,
-		appState:        appState,
-		wasmModuleState: &wasmGenesisState,
-	}, nil
+	return NewGenesisData(
+		genFile,
+		genDoc,
+		appState,
+		&wasmGenesisState,
+	), nil
+}
+
+type DefaultGenesisIO struct {
```

Comment:
Why is this here? Why not return `DefaultGenesisReader`?

I see that a project extending this would want to wrap it as you did below, but I don't see the need for it.

Reply:
`DefaultGenesisIO` implements `GenesisReader` and `GenesisMutator`. 
My motivation for the mutator was to give more flexibility for the genesis structure. In wasmd it is:
```json
{
"app_state":{
  "wasm":{
       "genesis":"data"
    }
}
```
With a different mutator implementation the structure can be different.

Reply:
Okay, so someone could inherit the `GenesisReader` but provide a custom `GenesisMutator`.

I agree on the two interfaces, it was just a thought that `&DefaultGenesisIO{GenesisReader: DefaultGenesisReader{}}` just delegates to the DefaultGenesisReader, so why not just return that. Another project could do this wrapping you do here.

Anyway, this is a super minor point. Feel free to merge as is if there is a reason I don't understand.

Reply:
Now I got it, sorry. Agree with you. :-)